content/docs/atom-guide/look-dev/tools/material-canvas.md

```diff
@@ -0,0 +1,302 @@
+                "defaultDataType": "float",
+                "settings": {
+                    "instructions": [
+                        "SLOTTYPE SLOTNAME = SLOTVALUE;"
+                    ]
+                }
+            }
+        ],
+        "outputSlots": [
+            {
+                "name": "outValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypeRegex": "float",
+                "defaultDataType": "float",
+                "settings": {
+                    "instructions": [
+                        "SLOTTYPE SLOTNAME = inValue;"
+                    ]
+                }
+            }
+        ]
+    }
+}
+```
+
+#### Material Graph Node Configuration Attributes
+Every node configuration must have a unique ID. This ensures that they are uniquely identifiable, regardless of location on disk, project, name collisions, or other factors.
+
+The node category determines how nodes will be grouped together in the node palette tree.
+
+The node title is the name displayed in the node palette and on the top of the node in the graph view. It will also be used to help create unique symbol names and variable names in shader code.
+
+The node title palette name is an optional field specifying which style sheet palette to use for the title bar on the node. Style sheets configure styling, coloring, fonts, and other attributes that control how elements in the node palette and graph view are displayed. Style sheets are defined in a separate, application wide file.
+
+The node description is an optional field that will be presented as a tool tip when hovering over a node in the palette.
+
+Slot data type groups contains a delimited list of slots names. The Material Canvas graph traversal and code generation process will enforce that all slot names listed in this field are promoted to the same data type, if they are compatible. Currently, if all of the listed slots reference scalar or vector values than all of the slot values will be promoted to the largest data type. For example, if all of the slots on the node are scalar values but an incoming connection is a three-dimensional vector then all of the other slots will be up converted to three dimensional vectors. This is necessary, in addition to other forms of casting, so that variables generated from different incoming types will be compatible with code and function calls defined in the node.
+
+#### Material Graph Node Slot Configuration Attributes
+Every slot configuration must have a unique name with respect to the node. The slot name is used to uniquely identify and address the slot, setup connections between slots, and is used to create a unique variable name in generated shader code and other files. The name will also be used as the display name in the UI if there is no specific display name specified. Changing the name will break connections and lose any data associated with the slot.
```

Comment:
```suggestion
Every slot configuration must have a unique name with respect to the node. The slot name is used to identify and address the slot, set up connections between slots, and create a unique variable name in generated shader code and other files. The name is also used as the display name in the UI, if there is no display name specified. Avoid changing the name because it breaks connections and loses any data associated with the slot.
```